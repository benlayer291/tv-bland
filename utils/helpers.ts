export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
