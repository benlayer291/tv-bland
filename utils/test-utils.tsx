import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
