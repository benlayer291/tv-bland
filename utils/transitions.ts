type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
