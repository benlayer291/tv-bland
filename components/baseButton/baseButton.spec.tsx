import React from 'react'
import { render, RenderResult, screen } from '../../utils/test-utils'
import userEvent from '@testing-library/user-event'

import BaseButton from './baseButton'

const onClick = jest.fn()
const propsDefault = {
  text: 'Text',
  onClick,
}

const init = (props?: Record<string, unknown>): RenderResult => {
  const propsInit = { ...propsDefault, ...props }

  return render(<BaseButton {...propsInit} />)
}

describe('<BaseButton />', () => {
  beforeEach(() => init())

  it('renders correctly with button text', () => {
    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveTextContent(propsDefault.text)
  })

  it('calls the onClick callback handler', async () => {
    await userEvent.click(screen.getByRole('button'))

    expect(onClick).toHaveBeenCalledTimes(1)
  })

  it('matches snapshot', () => {
    expect(screen.getByRole('button')).toMatchSnapshot()
  })
})
