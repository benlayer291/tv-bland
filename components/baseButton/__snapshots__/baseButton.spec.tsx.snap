// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton /> matches snapshot 1`] = `
<button
  class="BaseButton"
  type="button"
>
  Text
</button>
`;
